docs/pages/reference/backends.mdx

```diff
@@ -391,6 +391,62 @@ credentials, or [managed
 identity](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview)
 credentials.
 
+### Google Cloud IAM authentication
+
+If you are running Teleport on Google Cloud, Teleport can make use of [IAM
+Authentication](https://cloud.google.com/sql/docs/postgres/iam-authentication)
+to connect to an GCP Cloud SQL for PostgreSQL without having to manage any
+secrets:
+```yaml
+teleport:
+  storage:
+    type: postgresql
+    auth_mode: gcp-cloudsql
+
+    # GCP connection name is in format of <project>:<location>:<instance>.
+    gcp_connection_name: <Var name="project:location:instance"/>
+
+    # Type of IP for GCP connections. Valid options are:
+    # - "" (default to "public")
+    # - "public"
+    # - "private"
+    # - "psc" (for Private Service Connect)
+    gcp_ip_type: <Var name="public"/>
```

Comment:
Is this something users would need to set? How would they determine which one to pick?

Reply:
The owner of the Cloud SQL instance configures the IP mode when creating the database (or can modify it after)
<img width="546" alt="Screenshot 2024-06-05 at 11 57 33 AM" src="https://github.com/gravitational/teleport/assets/12417665/94fa3a1e-a8e3-46e0-8d84-17a921c5d6e0">


So here `gcp_ip_type` depends on what IPs DB has and which is accessible to Auth.